RevenueCatUI/CustomerCenter/View+PresentCustomerCenter.swift

```diff
@@ -0,0 +1,124 @@
+@available(macOS, unavailable, message: "RevenueCatUI does not support macOS yet")
+@available(tvOS, unavailable, message: "RevenueCatUI does not support tvOS yet")
+extension View {
+
+    /// Presents the ``CustomerCenter``.
+    /// Example:
+    /// ```swift
+    /// var body: some View {
+    ///    YourApp()
+    ///      .presentCustomerCenter()
+    /// }
+    /// ```
+
+    /// - Parameter isPresented: Binding indicating whether the customer center should be displayed
+    /// - Parameter onDismiss: Callback executed when the customer center wants to be dismissed.
+    /// Make sure you stop presenting the customer center when this is called
+    /// - Parameter presentationMode: The desired presentation mode of the customer center. Defaults to `.sheet`.
+    public func presentCustomerCenter(
+        isPresented: Binding<Bool>,
+        onDismiss: @escaping () -> Void,
+        presentationMode: CustomerCenterPresentationMode = .default
+    ) -> some View {
+        return self.modifier(PresentingCustomerCenterModifier(
+            isPresented: isPresented,
+            onDismiss: onDismiss,
+            myAppPurchaseLogic: nil,
+            presentationMode: presentationMode
+        ))
+    }
+
+}
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+@available(macOS, unavailable)
+@available(tvOS, unavailable)
+private struct PresentingCustomerCenterModifier: ViewModifier {
+
+    var presentationMode: CustomerCenterPresentationMode
+    var onDismiss: (() -> Void)
+
+    init(
+        isPresented: Binding<Bool>,
+        onDismiss: @escaping () -> Void,
+        myAppPurchaseLogic: MyAppPurchaseLogic?,
+        presentationMode: CustomerCenterPresentationMode,
+        purchaseHandler: PurchaseHandler? = nil
+    ) {
+        self._isPresented = isPresented
+        self.presentationMode = presentationMode
+        self.onDismiss = onDismiss
+        self._purchaseHandler = .init(wrappedValue: purchaseHandler ??
+                                      PurchaseHandler.default(performPurchase: myAppPurchaseLogic?.performPurchase,
+                                                              performRestore: myAppPurchaseLogic?.performRestore))
+    }
+
+    @StateObject
+    private var purchaseHandler: PurchaseHandler
+
+    @Binding
+    var isPresented: Bool
+
+    func body(content: Content) -> some View {
+        Group {
+            switch presentationMode {
+            case .sheet:
+                content
+                    .sheet(isPresented: self.$isPresented, onDismiss: self.onDismiss) {
+                        self.customerCenterView()
+                    }
+            case .fullScreen:
+                content
+                    .fullScreenCover(isPresented: self.$isPresented, onDismiss: self.onDismiss) {
+                        self.customerCenterView()
+                    }
+            }
+        }
+    }
+
+    private func customerCenterView() -> some View {
+        CustomerCenterView()
```

Comment:
We need to allow passing in the `purchaseHandler` (and a lot of other things to the `CustomerCenterView`). We can add those in separate PRs.